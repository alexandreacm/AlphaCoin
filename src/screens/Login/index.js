import React, { useState, useContext } from 'react';
import { TextInput } from 'react-native';

import {
  StyledContainer,
  StyledLabel as Label,
  StyledContainerImage,
  StyledImage,
  StyledViewAreaInput,
  StyledButtonLogin,
  StyledIcon,
  StyledTextButtonLogin,
  StyledIconUser,
  StyledInput
} from './styles';

import colors from '@/theme/colors';
import icLogo from '@/assets/images/logo.jpg';

import { AuthContext } from '@/contexts/AuthProvider';

export default function Login() {
  const [userName, setUserName] = useState();
  const { SignIn } = useContext(AuthContext);

  return (
    <StyledContainer>
      <StyledContainerImage>
        <StyledImage source={icLogo} width={150} height={150} />
      </StyledContainerImage>
      <StyledViewAreaInput>
        <StyledIconUser>
          <StyledIcon name='user' size={20} color={colors.PLACE_HOLDER} />
        </StyledIconUser>
        <StyledInput>
          <TextInput
            value={userName}
            onChangeText={setUserName}
            placeholder='User name...'
            placeholderTextColor={colors.PLACE_HOLDER}
          />
        </StyledInput>
      </StyledViewAreaInput>
      <StyledButtonLogin
        onPress={() => {
          SignIn(userName);
        }}
      >
        <StyledTextButtonLogin>Login</StyledTextButtonLogin>
      </StyledButtonLogin>
    </StyledContainer>
  );
}
