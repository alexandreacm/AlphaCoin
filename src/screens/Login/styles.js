import styled from 'styled-components/native';

import useFontFamily from '@/hooks/useFontFamily';
import { handlePixels } from '@/helpers/functions/utils';

import colors from '@/theme/colors';
import Icon from 'react-native-vector-icons/Feather';

const StyledContainer = styled.View`
  flex: 1;
  background-color: ${colors.BG};
  justify-content: center;
  align-items: center;
  padding: 10px;
`;

const StyledContainerImage = styled.View`
  width: 60px;
  height: 60px;
  background-color: ${colors.PRIMARY};
  justify-content: center;
  align-items: center;
`;

const StyledImage = styled.Image`
  width: 100px;
  height: 100px;
  border-radius: 50px;
`;

const StyledButtonLogin = styled.TouchableOpacity`
  width: 100%;
  padding: 10px;
  background: ${colors.PRIMARY};
  justify-content: center;
  align-items: center;
  border-radius: 8px;
`;

const StyledViewAreaInput = styled.View`
  width: 100%;
  height: 50px;
  background: ${colors.BG};
  border-radius: 10px;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  border-width: 1px;
  border-color: ${colors.BLACK};
  margin-top: 50px;
  margin-bottom: 15px;
`;

const StyledLabel = styled.Text`
  color: ${({ color }) => color};
  text-align: ${({ textAlign }) => (textAlign ? textAlign : 'center')};
  font-size: ${({ fontSize }) => handlePixels(fontSize)};
  font-family: ${({ fontWeight }) => useFontFamily(fontWeight)};
  line-height: ${({ lineHeight }) =>
    lineHeight ? handlePixels(lineHeight) : 0};
  margin-bottom: ${({ marginBottom }) =>
    marginBottom ? handlePixels(marginBottom) : 0};
`;

const StyledTextButtonLogin = styled.Text`
  color: ${colors.WHITE};
  text-align: center;
  font-size: 18px;
  font-family: ${useFontFamily(700)};
`;

const StyledIcon = styled(Icon)`
  margin-left: 10px;
`;

const StyledIconUser = styled.View`
  width: 12%;
  height: 50px;
  justify-content: center;
  align-items: flex-start;
`;
const StyledInput = styled.View`
  width: 88%;
  height: 50px;
`;

export {
  StyledContainer,
  StyledLabel,
  StyledContainerImage,
  StyledImage,
  StyledViewAreaInput,
  StyledIcon,
  StyledButtonLogin,
  StyledTextButtonLogin,
  StyledIconUser,
  StyledInput
};
