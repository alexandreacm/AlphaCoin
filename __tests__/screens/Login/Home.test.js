import 'react-native';
import React from 'react';
import {
  matchSnapshotWithTheme,
  renderWithTheme
} from '@/helpers/functions/testing';

import Login from '@/screens/Login';

describe('tests in Login Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<Login />);
  });

  test('renders correctly', () => {
    renderWithTheme(<Login />);
  });
});
