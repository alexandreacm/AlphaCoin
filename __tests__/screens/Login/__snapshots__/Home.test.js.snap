// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in Login Screen matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#f3f3f3",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#323b4a",
          "height": 60,
          "justifyContent": "center",
          "width": 60,
        },
      ]
    }
  >
    <Image
      height={150}
      source={
        Object {
          "testUri": "../../../src/assets/images/logo.jpg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 50,
            "height": 100,
            "width": 100,
          },
        ]
      }
      width={150}
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#f3f3f3",
          "borderColor": "#000",
          "borderRadius": 10,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "flex-start",
          "marginBottom": 15,
          "marginTop": 50,
          "width": "100%",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "height": 50,
            "justifyContent": "center",
            "width": "12%",
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#808080",
              "fontSize": 20,
            },
            Array [
              Object {
                "marginLeft": 10,
              },
            ],
            Object {
              "fontFamily": "Feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        ïˆ„
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "height": 50,
            "width": "88%",
          },
        ]
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="User name..."
        placeholderTextColor="#808080"
      />
    </View>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#323b4a",
        "borderRadius": 8,
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFF",
            "fontFamily": "Poppins-Bold",
            "fontSize": 18,
            "textAlign": "center",
          },
        ]
      }
    >
      Login
    </Text>
  </View>
</View>
`;
